### PR TITLE
New Doc: Adding a suggestion supplier to a plugin

### PR DESCRIPTION
**Pull request by briangann**
**What this PR does / why we need it**:

Adds docs for creating a suggestion supplier for a plugin

**Which issue(s) this PR fixes**:

Fixes # https://github.com/grafana/grafana/issues/84419

### CODE REVIEW

File: docusaurus/docs/migration-guides/angular-react/suggestion-supplier.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  - Angular
+  - migration
+  - suggestion
+---
+
+# Angular to React: Add a suggestion supplier
+
+You can add a suggestion supplier to examine query data coming from a panel and suggest usage of the plugin for the type of data detected. This guide provides instructions for doing so along with links to relevant examples.
+
+A single stat panel would be ranked "high" for a single series, and "low" for multiple series (or even none).
```
Comment (josmperez) [inline docusaurus/docs/migration-guides/angular-react/suggestion-supplier.md:21]: This sentence seems disconnected from what came before. Can you clarify?

File: docusaurus/docs/migration-guides/angular-react/suggestion-supplier.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    list.append({
+      name: 'MyPanel',
+    });
+  }
+}
+```
+
+:::note
+
+When creating a suggestion supplier be certain the plugin can actually render something for the data being provided.
```
Comment (josmperez) [inline docusaurus/docs/migration-guides/angular-react/suggestion-supplier.md:66]: What happens if you don't?

File: docusaurus/docs/migration-guides/angular-react/suggestion-supplier.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+:::
+
+## Additional resources
+
+Reference these suggestion suppliers to get ideas for further customization:
```
Comment (josmperez) [inline docusaurus/docs/migration-guides/angular-react/suggestion-supplier.md:72]: Can you add a sentence or part of a sentence to note what's distinctive about the suggestion suppliers in each of these plugins?

**Review by josmperez — APPROVED** [review]
I've made some editorial commits and suggestions, but no blocking issues. LGTM.

File: docusaurus/docs/migration-guides/angular-react/suggestion-supplier.md (hunk 1 above)
Comment (briangann) [reply docusaurus/docs/migration-guides/angular-react/suggestion-supplier.md:21]: will add text here, i left out context

File: docusaurus/docs/migration-guides/angular-react/suggestion-supplier.md (hunk 2 above)
Comment (briangann) [reply docusaurus/docs/migration-guides/angular-react/suggestion-supplier.md:66]: Grafana will either display a blank panel and/or an error message will be displayed. It's best to offer the plugin only to query data that matches well-known criteria that the author can validate.

File: docusaurus/docs/migration-guides/angular-react/suggestion-supplier.md (hunk 3 above)
Comment (briangann) [reply docusaurus/docs/migration-guides/angular-react/suggestion-supplier.md:72]: will do!

**Comment by briangann** [comment]
updated with more details, hopefully this reads a little better

**Pull request merged**